### PR TITLE
code improvements

### PR DESCRIPTION
**Pull request by sinisastanic**

### CODE REVIEW

File: src/debugtopus.ts [hunk 1]
```diff
@@ -4,8 +4,9 @@ import { promisify } from "util";
 import { exec } from "child_process";
 import { randomUUID } from "crypto";
 import path from "path";
-
 import { getPlaywrightCode } from "./octomind-api";
+// eslint-disable-next-line import/no-commonjs
```
Comment (sinisastanic) [inline src/debugtopus.ts:8]: @Germandrummer92

File: src/debugtopus.ts [hunk 2]
```diff
@@ -42,11 +43,9 @@ export const prepareTestRun = async ({
 }> => {
   const code = await getPlaywrightCode(testId, token, url, octomindUrl);
 
-  const dirname = __dirname;
-  let tempDir = path.join(dirname, "..", "..", "..", "..", "temp");
-  if (process.env.NODE_ENV === "test") {
-    tempDir = path.join(dirname, "..", "temp");
-  }
+  // @ts-ignore
```
Comment (sinisastanic) [inline src/debugtopus.ts:46]: @Germandrummer92

File: src/debugtopus.ts (hunk 1 above)
Comment (Germandrummer92) [reply src/debugtopus.ts:8]: just do a normal import

import path, { dirname} from 'path';

File: src/debugtopus.ts (hunk 2 above)
Comment (Germandrummer92) [reply src/debugtopus.ts:46]: not sure if we can solve this, maybe 

```ts
//ts-expect-error require.main.filename is not typed
```

**Review by Germandrummer92 — APPROVED** [review]

File: src/debugtopus.ts [hunk 3]
```diff
@@ -25,6 +24,28 @@ export default defineConfig({
...
+  }
+
+  if (infiniteLoopPrevention === 0) {
+    throw new Error("can't find root level node modules :/");
+  }
+
+  const tempDir = path.join(appDir, "temp");
+  return tempDir;
+};
+
```
Comment (sinisastanic) [inline src/debugtopus.ts:48]: would love to unit test this, just no idea how @Germandrummer92

File: src/debugtopus.ts [hunk 4]
```diff
@@ -42,13 +63,14 @@ export const prepareTestRun = async ({
...
-  let tempDir = path.join(dirname, "..", "..", "..", "..", "temp");
-  if (process.env.NODE_ENV === "test") {
-    tempDir = path.join(dirname, "..", "temp");
+  const nodeModule = require.main;
+  if (!nodeModule) {
+    throw new Error("package was not installed as valid nodeJS module");
   }
-
+  const appDir = dirname(nodeModule.filename);
+  const tempDir = getTempDirOnPackageRootLevel(appDir);
```
Comment (sinisastanic) [inline src/debugtopus.ts:71]: @Germandrummer92 is this okay method name?

File: src/debugtopus.ts (hunk 3 above)
Comment (sinisastanic) [reply src/debugtopus.ts:48]: somehow mocking `existsSync` to return false X amount of time and then true 🤷

File: src/debugtopus.ts (hunk 4 above)
Comment (Germandrummer92) [reply src/debugtopus.ts:70]: i like, yes!

File: src/debugtopus.ts (hunk 3 above)
Comment (Germandrummer92) [reply src/debugtopus.ts]: yes thats probably the right idea, something like this should work:

```ts
jest.mock("fs");

let callCount = 0;
jest.mocked(existsSync).mockImplementation(() => {
    callCount++;
    if (callCount === 5) {
        return true;
    } 
    return false;
});
```

**Pull request merged**